src/transformers/__init__.py

```diff
@@ -1780,7 +1780,9 @@
             "MPNetPreTrainedModel",
         ]
     )
-    _import_structure["models.mt5"].extend(["MT5EncoderModel", "MT5ForConditionalGeneration", "MT5Model"])
+    _import_structure["models.mt5"].extend(
+        ["MT5EncoderModel", "MT5ForConditionalGeneration", "MT5Model", "MT5PreTrainedModel", "MT5Stack"]
```

Comment:
The stack is an internal class. It can be in the intermediate init of mt5 but shouldn't be in the main init.
```suggestion
        ["MT5EncoderModel", "MT5ForConditionalGeneration", "MT5Model", "MT5PreTrainedModel"]
```

Reply:
done